### ISSUE TITLE
guidance/doc when -Weverything is unrecognized

### ISSUE THREAD

**Issue by roycewilliams** [alaskacommunications/akcom-udpecho#2]
Might need a tweak to minimum versions, or syntax, or just a FAQ item for the README?
```
libtool --mode=compile --tag=CC gcc -O2 -std=gnu11 -pedantic -W -Wall -Wextra -Weverything -Wno-unknown-pragmas -Wno-format-nonliteral -Wno-reserved-id-macro -Wno-unused-macros -DPACKAGE_VERSION='"0.2.0.gd2084ce"' -o akcom-udpecho.lo -c akcom-udpecho.c
libtool: compile:  gcc -O2 -std=gnu11 -pedantic -W -Wall -Wextra -Weverything -Wno-unknown-pragmas -Wno-format-nonliteral -Wno-reserved-id-macro -Wno-unused-macros -DPACKAGE_VERSION=\"0.2.0.gd2084ce\" -c akcom-udpecho.c  -fPIC -DPIC -o .libs/akcom-udpecho.o
gcc: error: unrecognized command line option '-Weverything'
make: *** [GNUmakefile:76: akcom-udpecho.lo] Error 1
```
On the target system, I'm at or above the current stated minimum versions:
```
$ for bin in gcc libtool autoconf automake git; do ${bin} --version | head -n 1; done
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
libtool (GNU libtool) 2.4.6
autoconf (GNU Autoconf) 2.69
automake (GNU automake) 1.16.1
git version 2.25.1
```

**Comment by pyromiko** [comment]
same issue:

```
/usr/src/akcom-udpecho/src# make
libtool --mode=compile --tag=CC gcc -O2 -std=gnu11 -pedantic -W -Wall -Wextra -Weverything -Wno-unknown-pragmas -Wno-format-nonliteral -Wno-reserved-id-macro -Wno-unused-macros -DPACKAGE_VERSION='"0.2.0.gd2084ce"' -o akcom-udpecho.lo -c akcom-udpecho.c
libtool: compile:  gcc -O2 -std=gnu11 -pedantic -W -Wall -Wextra -Weverything -Wno-unknown-pragmas -Wno-format-nonliteral -Wno-reserved-id-macro -Wno-unused-macros -DPACKAGE_VERSION=\"0.2.0.gd2084ce\" -c akcom-udpecho.c  -fPIC -DPIC -o .libs/akcom-udpecho.o
gcc: error: unrecognized command line option â€˜-Weverythingâ€™
make: *** [GNUmakefile:76: akcom-udpecho.lo] Error 1
```


```
gcc (Debian 8.3.0-6) 8.3.0
libtool (GNU libtool) 2.4.6
autoconf (GNU Autoconf) 2.69
automake (GNU automake) 1.16.1
git version 2.20.1
```

**Comment by syzdek** [comment]
-Weverything is a CLANG option.  I've removed -Wextra and -Weverything from the GNUmakefile in the source directory (commit 475ddb90d8a40b2bf9eea3dd9125825404056a07).  The GNUmakefile is legacy from when I started the project and may go away if _akcom-echoplusd_ is updated to service concurrent packets when utilizing the the __drop__ and __delay__ feature.  I've updated the order of the build in the README and added a notice to the GNUmakefile stating the  developer's preference (commit 8b96dba7cf9ba3c7b108980eb4f3cfa26806b792). I have also updated the software version listed in the readme (commit 9b7da228b9b22f6273e096e2be2324c89dca888f).

**Comment by syzdek** [comment]
I also simplified the GNUmakefile by removing GNU libtool and relying on GNU Make's internal compiler rules (commit fb67debbfe9ea7cd4e7acb5a1456061d462a0614).  The changes have been tested on MacOS 11.6 (x86_64), Slackware 14.2 (x86_64), and Alpine Linux 3.13 (amd64).

**Comment by roycewilliams** [comment]
That was quick, thanks! Now getting:
```
$ pwd
/usr/local/src/net/akcom-udpecho/build

$ make
make  all-recursive
make[1]: Entering directory '/usr/local/src/net/akcom-udpecho/build'
make[2]: Entering directory '/usr/local/src/net/akcom-udpecho/build'
  CC       src/akcom_udpecho-akcom-udpecho.o
  CCLD     src/akcom-udpecho
  CC       src/akcom_udpechod-akcom-udpechod.o
../src/akcom-udpechod.c: In function 'my_daemonize':
../src/akcom-udpechod.c:526:7: error: ignoring return value of 'fscanf', declared with attribute warn_unused_result [-Werror=unused-result]
  526 |       fscanf(fs, "%i", &pid);
      |       ^~~~~~~~~~~~~~~~~~~~~~
../src/akcom-udpechod.c:720:4: error: ignoring return value of 'write', declared with attribute warn_unused_result [-Werror=unused-result]
  720 |    write(fd, buff, strlen(buff));
      |    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [Makefile:1258: src/akcom_udpechod-akcom-udpechod.o] Error 1
make[2]: Leaving directory '/usr/local/src/net/akcom-udpecho/build'
make[1]: *** [Makefile:1421: all-recursive] Error 1
make[1]: Leaving directory '/usr/local/src/net/akcom-udpecho/build'
make: *** [Makefile:979: all] Error 2
```

**Comment by syzdek** [comment]
Ironically, I saw today that someone in China had the same issue ([Technical Notes from an Engineering Lion](https://engineer--leo-blogspot-com.translate.goog/2021/02/udp-echo.html?_x_tr_sch=http&_x_tr_sl=zh-TW&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=sc)).  This should be resolved in commits 1ea945e00272363b676e9bec41dbedc619acfeaf and daa7be7df8e3bf2937115dede7cd1708265dbe8f.

**Comment by roycewilliams** [comment]
Client up and running - awesome, thanks!
```
$ akcom-udpecho -c 5 --rfc 209.112.131.108 30006
UDPECHO 209.112.131.108:30006 (209.112.131.108:30006): 40 bytes
udpecho_seq=1 time=20.9 ms
udpecho_seq=2 time=11.6 ms
udpecho_seq=3 time=12.8 ms
udpecho_seq=4 time=16.3 ms
udpecho_seq=5 time=12.1 ms

--- 209.112.131.108 udpecho statistics ---
5 packets transmitted, 5 packets received, 0.0% packet loss
round-trip min/avg/max = 11.6/14.7/20.9 ms
```
